File: exercism/csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    private static int Number { get; set; }

    private static string Check(int divisor, string word)
        => Number % divisor == 0 ? word : String.Empty;

    public static string Convert(int number)
    {
        Number = number;

        var txt = String.Empty;
       
        txt += Check(3, "Pling");
        txt += Check(5, "Plang");
        txt += Check(7, "Plong");
        
        return string.IsNullOrWhiteSpace(txt)
            ? number.ToString()
            : txt;
    }
}

File: exercism/python/raindrops/raindrops.py
def convert(number):
    factor_sounds = {
        3: 'Pling',
        5: 'Plang',
        7: 'Plong'
    }

    raindrops = ''
    for factor, sound in factor_sounds.items():
        raindrops += sound if number % factor == 0 else ''

    return raindrops if raindrops else str(number)
